### DISCUSSION TITLE
I want to print 1 or 0 on the bottom of each wave form

### DISCUSSION THREAD

**Discussion by quantrpeter** [OpenHantek/OpenHantek6022#393] [Q&A]
Hi
   I want to print 1 or 0 on the bottom of each wave form. Please tell me where i should modify? Should i just add my code to void GlScope::drawVoltageChannelGraph ?  How to get the value of each point? I can't dump out all values from these:

Graph::VaoCount &v = graph.vaoVoltage[ channel ];

thanks
Peter

**Comment by quantrpeter** [comment]
ok, I can dump out all points in wave form here. Please tell me what to do next, thanks

![image](https://github.com/OpenHantek/OpenHantek6022/assets/50471577/1803e3be-f58e-434d-9081-a530758cb5a2)

**Comment by Ho-Ro** [comment]
Please explain what you want to achieve. If you want the measured values as a list you can export them as csv or json from the export menu. If you want to hack your own format take this as a starter.

Also reading [docs/developer_info.md](https://github.com/OpenHantek/OpenHantek6022/blob/main/docs/developer_info.md) could give you some understanding of the data flow. On remark - the generating of the `MATH` channel was enhanced to support more functions and was shifted to [hantekdso/mathchannel.cpp](https://github.com/OpenHantek/OpenHantek6022/blob/main/openhantek/src/hantekdso/mathchannel.cpp) to become also a valid trigger source. I will update this documentation accordingly for the upcoming 3.4 release.
